core/io/image.cpp

```diff
@@ -2182,7 +2183,7 @@ void Image::initialize_data(int p_width, int p_height, bool p_use_mipmaps, Forma
 	ERR_FAIL_COND_MSG(p_height > MAX_HEIGHT,
 			vformat("The Image height specified (%d pixels) cannot be greater than %d pixels.", p_height, MAX_HEIGHT));
 	ERR_FAIL_COND_MSG(p_width * p_height > MAX_PIXELS,
-			vformat("Too many pixels for Image. Maximum is %dx%d = %d pixels.", MAX_WIDTH, MAX_HEIGHT, MAX_PIXELS));
+			vformat("Too many pixels for Image. Maximum is %dx%d = %d pixels.", MAX_TEXTURE_SIZE, MAX_TEXTURE_SIZE, MAX_PIXELS));
```

Comment:
This will be misleading, `MAX_TEXTURE_SIZE * MAX_TEXTURE_SIZE` only make sense for square image, probably better to remove `%dx%d` from all these messages or list dimensions.
```
Maximum width is MAX_WIDTH, maximum height is MAX_HEIGHT, maximum total number of pixes is MAX_PIXELS.
```